### PR TITLE
Support extraction of ids for more providers (L - Z) found in data

### PR DESCRIPTION
**Pull request by shashank025**
| Key Details |
|-|
| Resolves: #254 |
State: |
Site: |

## Notes

Continuation of work started with #332. This PR builds on https://github.com/CAVaccineInventory/vaccine-feed-ingest-schema/pull/21 to add support of extraction of more vaccine providers.

This PR also refactors the provider_id code to make it easier to maintain and extend by extract config out of code.

## Data sample

I will run the runners shortly ...

## Before Opening a PR
- [X] I tested this using the CLI (e.g., `poetry run vaccine-feed-ingest <state>/<site>`)
- [X] I ran auto-formatting: `poetry run tox -e lint-fix`

### CODE REVIEW

File: vaccine_feed_ingest/utils/normalize.py [hunk 1]
```diff
@@ -7,142 +7,118 @@
 import url_normalize
 from vaccine_feed_ingest_schema.location import VaccineProvider
 
+VACCINE_PROVIDER_REGEXES = {
+    VaccineProvider.ALBERTSONS: {
```
Comment (bryanculbertson) [inline vaccine_feed_ingest/utils/normalize.py:11]: I am concerned about this data model. This process combines brands that are run by the same corporation into the same id space. Are you sure that is the way it works in practice?

By this I mean, is Albertsons #5301 and Safeway #5301 the same store, or are they different stores?

Comment (bryanculbertson) [reply vaccine_feed_ingest/utils/normalize.py:11]: What do you think of extracting the provider, brand, and id and creating an ID from those parts to ensure uniqueness e.g. `albertsons:osco_5301` or `albertsons_osco:5301`

Comment (shashank025) [reply vaccine_feed_ingest/utils/normalize.py:11]: I don't know enough about either our front-end, or how the data is used to truly weigh in, but after reviewing:
https://www.cdc.gov/vaccines/covid-19/retail-pharmacy-program/participating-pharmacies.html - my suggestion is that we enhance the schema to explicitly represent brands, providers, and the relationship between them.

To be more specific:

* The `VaccineProvider` enum hereafter represents individual brands (e.g., Big Y Pharmacy, Brookshires Pharmacy, etc)
* A new enum, say: `VaccineProviderParent` (with values like Topco, Kroger, etc) represents businesses that have one or more `VaccineProvider` subsidiaries
* Add a `getParent()` helper method to  the `VaccineProvider` enum and implement it appropriately to return an `Optional[VaccineProviderParent]`.
* Modify `provider_id_from_name()` to return either a `VaccineProvider` or `VaccineProviderParent` (use the `Union` type).

Comment (shashank025) [reply vaccine_feed_ingest/utils/normalize.py:11]: Explicitly representing brands, parents, and their relationship in the schema will make it easier to build and maintain richer frontends showing users these relationships (if necessary).

Comment (shashank025) [reply vaccine_feed_ingest/utils/normalize.py:11]: Tagging @nschiefer to weigh in since he created #254

Comment (shashank025) [reply vaccine_feed_ingest/utils/normalize.py:11]: Per discussion on Discord, the above proposal is fine, but the adding of the parent enum etc can be done later. For now, this PR will proceed with the assumption that `VaccineProvider` represents individual brands.

Comment (shashank025) [reply vaccine_feed_ingest/utils/normalize.py]: Done.

**Comment by shashank025** [comment]
The broken tests will be fixed once the repo is updated with https://github.com/CAVaccineInventory/vaccine-feed-ingest-schema/releases/tag/1.2.2

**Review by bryanculbertson — APPROVED** [review]
Thank you!

**Pull request merged**